caffe2/CMakeLists.txt

```diff
@@ -211,10 +206,579 @@ if(NOT BUILD_ATEN_ONLY)
+      install(TARGETS thnvrtc DESTINATION "${TORCH_INSTALL_LIB_DIR}")
+
+    endif()
+  endif ()
+
+  if (USE_CUDA)
+    list(APPEND Caffe2_GPU_SRCS
+      ${TORCH_SRC_DIR}/csrc/autograd/profiler_cuda.cpp
+      ${TORCH_SRC_DIR}/csrc/autograd/functions/comm.cpp
+      ${TORCH_SRC_DIR}/csrc/cuda/comm.cpp
+    )
+  endif()
+
+  if (NOT NO_API)
+    list(APPEND TORCH_SRCS
+      ${TORCH_SRC_DIR}/csrc/api/src/cuda.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/datasets/mnist.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/samplers/distributed.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/samplers/random.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/samplers/sequential.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/data/samplers/stream.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/jit.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/init.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/module.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/batchnorm.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/conv.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/dropout.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/embedding.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/functional.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/linear.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/named_any.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/nn/modules/rnn.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/adagrad.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/adam.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/lbfgs.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/optimizer.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/rmsprop.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/serialize.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/optim/sgd.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/serialize/input-archive.cpp
+      ${TORCH_SRC_DIR}/csrc/api/src/serialize/output-archive.cpp
+    )
+  endif()
+
+
+  list(APPEND Caffe2_CPU_SRCS ${TORCH_SRCS})
+endif()
+
+# ==========================================================
+# END formerly-libtorch sources
+# ==========================================================
+
+
+
 # Compile exposed libraries.
 add_library(caffe2 ${Caffe2_CPU_SRCS})
+
+
+
+# ==========================================================
+# formerly-libtorch flags
+# ==========================================================
+
+if (NOT BUILD_ATEN_MOBILE)
+
+  # Forces caffe2.pb.h to be generated before its dependents are compiled.
+  # Adding the generated header file to the ${TORCH_SRCS} list is not sufficient
+  # to establish the dependency, since the generation procedure is declared in a different CMake file.
+  # See https://samthursfield.wordpress.com/2015/11/21/cmake-dependencies-between-targets-and-files-and-custom-commands/#custom-commands-in-different-directories
+  add_dependencies(caffe2 Caffe2_PROTO)
+
+  target_compile_definitions(caffe2 PUBLIC _THP_CORE)
+
+
+  # until they can be unified, keep these lists synced with setup.py
```

Comment:
probably it's worth to look into it in the future (not sure what exact history of this duplication is) - from setup.py it seems we're building only stub.cpp, so it might be outdated comment